thoth-api/src/model/mod.rs

```diff
@@ -528,6 +528,43 @@ impl Convert for f64 {
     }
 }
 
+/// Assign the leading domain of an identifier
+pub trait UrlIdentifier {
```

Comment:
Nice! There may be a few other places in existing code where we could use this, although that can wait for a future enhancement.